crates/wiggle/src/lib.rs

```diff
@@ -542,6 +542,26 @@ impl<'a, T> GuestPtr<'a, [T]> {
     pub fn as_ptr(&self) -> GuestPtr<'a, T> {
         GuestPtr::new(self.mem, self.offset_base())
     }
+
+    pub fn get(&self, index: u32) -> Option<GuestPtr<'a, T>> {
+        if index < self.len() {
+            Some(GuestPtr::new(self.mem, self.offset_base() + index))
+        } else {
+            None
+        }
+    }
+
+    pub fn get_range(&self, r: std::ops::Range<u32>) -> Option<GuestPtr<'a, [T]>> {
+        let range_length = r.end - r.start;
+        if r.start <= self.len() && r.end <= self.len() {
```

Comment:
I think this may also want to include the condition that `r.start <= r.end`?

Reply:
I was under the mistaken impression that this was guaranteed by Range construction, I'll correct that